### PR TITLE
Feature:AppBar-Display user email when is authenticated

### PR DESCRIPTION
**Pull request by SilviaTrujillano23**
## Describe your changes
 
- Add AuthContext and Provider to store user data
- Render event description from endpoint 
- Remove unused code

## Issue ticket number and link
https://trello.com/c/rHezN0Wq

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update: 
- [ ] Create storybook for components involved

AppBar when user is logged in

Screenshot 2023-07-04 at 23 46 21

### CODE REVIEW

File: src/contexts/Auth/AuthProvider.tsx [hunk 1]
```diff
@@ -12,15 +12,18 @@ interface VerifyApiResponse {
 }
 
 export const AuthProvider = ({ children }: Props) => {
-  let initialState = { isAuth: false, email: '', loadingIsAuth: true }
+  let initialState = { isAuth: false, user: {} }
   const api = Authentication()
 
   const verifyUser = async () => {
     try {
       const verifyApiResponse: VerifyApiResponse = await api.verifyAuth()
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:20]: Use destructuring
```suggestion
      const { isAuth,  user }: VerifyApiResponse = await api.verifyAuth()
```

File: src/contexts/Auth/AuthProvider.tsx [hunk 2]
```diff
@@ -12,15 +12,18 @@ interface VerifyApiResponse {
...
       dispatch({
         type: 'UPDATE_IS_AUTH',
-        payload: { isAuth: verifyApiResponse.isAuth },
+        payload: {
+          isAuth: verifyApiResponse.isAuth,
+          user: verifyApiResponse.user,
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:25]:
```suggestion
          isAuth,
          user,
```

File: src/contexts/Auth/AuthReducer.tsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 }
 
 export const reducer = (state: State, action: Action) => {
   switch (action.type) {
     case 'UPDATE_IS_AUTH':
-      return { ...state, isAuth: action.payload.isAuth }
+      return {
+        ...state,
+        isAuth: action.payload.isAuth,
+        user: action.payload.user,
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthReducer.tsx:16]: DRY! please use destructure

Comment (SilviaTrujillano23) [reply src/contexts/Auth/AuthReducer.tsx]: Done

File: src/contexts/Auth/AuthProvider.tsx (hunk 2 above)
Comment (SilviaTrujillano23) [reply src/contexts/Auth/AuthProvider.tsx]: Done, thanks :)

File: src/contexts/Auth/AuthProvider.tsx (hunk 1 above)
Comment (SilviaTrujillano23) [reply src/contexts/Auth/AuthProvider.tsx]: Done, thanks :)

File: src/api/events.js [hunk 4]
```diff
@@ -24,6 +25,10 @@ export default class Events extends baseRequest {
     const response = await this.get(`${this.methodAll}/${id}`)
     return response.data.data
   }
+  verifyUserEventSubscribed = async (id) => {
+    const response = await this.get(`${this.userMethod}/${this.method}/${id}`)
+    return response.data.data
```
Comment (jlora-mck) [inline src/api/events.js:30]: Destructure please
```suggestion
    const response = await this.get(`${this.userMethod}/${this.method}/${id}`)
    return response.data.data
```

File: src/contexts/Auth/AuthProvider.tsx [hunk 5]
```diff
@@ -2,33 +2,53 @@ import { useReducer, ReactNode } from 'react'
...
+    })
+  }
+  const getUserInfo = async (userUid: string) => {
+    //TODO: Add endpoint to access user data only
+  }
   const [state, dispatch] = useReducer(reducer, initialState)
+
   return (
-    <AuthContext.Provider value={{ state, verifyUser }}>
+    <AuthContext.Provider value={{ state, verifyUser, setLoginData }}>
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:51]: be carefull with adding to much here! what we usually do is one single method and in this method we do switch case

File: src/pages/Events/Events.tsx [hunk 6]
```diff
@@ -8,6 +8,7 @@ import NavigationWrapper from '../../components/Navigation/NavigationWrapper'
 import { HeadquarterAPI, ConferenceAPI } from '../../shared/api'
 import { Conference, Headquarter } from '../../shared/entities'
 import { sortAscending } from '../../tools/sorting'
+//import { useAuth } from '../../hook/useAuth'
```
Comment (jlora-mck) [inline src/pages/Events/Events.tsx:11]: please remove

Comment (SilviaTrujillano23) [reply src/pages/Events/Events.tsx:11]: Done

File: src/api/events.js (hunk 4 above)
Comment (SilviaTrujillano23) [reply src/api/events.js:30]: Done

File: src/contexts/Auth/AuthProvider.tsx (hunk 5 above)
Comment (SilviaTrujillano23) [reply src/contexts/Auth/AuthProvider.tsx:50]: taking note, next PR will add this change

**Review by jlora-mck — APPROVED** [review]

File: src/contexts/Auth/AuthProvider.tsx [hunk 7]
```diff
@@ -26,9 +26,9 @@ export const AuthProvider = ({ children }: Props) => {
           userUid,
         },
       })
-      if (isAuth) {
-        getUserInfo(userUid)
-      }
+      // if (isAuth) {
+      //   getUserInfo(userUid)
+      // }
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:31]: Add a todo before please, so we know why is this commented

File: src/contexts/Auth/AuthProvider.tsx [hunk 8]
```diff
@@ -43,7 +43,7 @@ export const AuthProvider = ({ children }: Props) => {
     })
   }
   //TODO: Add endpoint to access user data only
-  const getUserInfo = async (userUid: string) => {}
+  //const getUserInfo = async (userUid: string) => {}
```
Comment (jlora-mck) [inline src/contexts/Auth/AuthProvider.tsx:46]: same here

**Review by jlora-mck — APPROVED** [review]

**Review by FelipeLosada1996 — APPROVED** [review]

**Review by anversa-pro — APPROVED** [review]
🤯

**Pull request merged**